### PR TITLE
add sf method for as.geojson

### PR DESCRIPTION
**Pull request by cpsievert**

### CODE REVIEW

**Comment by sckott** [comment]
thanks @cpsievert - curious what the use case is that motivated this?

since it uses `sf`, i think we'd need to put `sf` in Suggests. This pkg itself doesn't depend yet on any of the pkgs with heavy system dependencies (rgeos, rgdal) so I'd prefer to put `sf` in Suggests.  

Does `sf::as_Spatial` cover all cases of objects with class `sf`?  are the other sf classes `sfg`, etc. covered by this method?

**Comment by cpsievert** [comment]
> curious what the use case is that motivated this?

I'm designing an htmlwidget for a client where the front-end (i.e., JavaScript) consumes geojson. I'm using `as.geojson()` to make sure the user-provided R object is of the right type before translating to JSON. By supporting `sf` objects the user will have more flexibility as to what they feed into the function.

> Does sf::as_Spatial cover all cases of objects with class sf?

I believe so. The man page says it supports any of the following classes -- sf, sfc_POINT, sfc_MULTIPOINT, sfc_LINESTRING, sfc_MULTILINESTRING, sfc_POLYGON, or sfc_MULTIPOLYGON.

**Comment by sckott** [comment]
okay, thanks for the background.

File: R/sp_to_geojson.R [hunk 1]
```diff
@@ -85,3 +85,9 @@ setMethod("as.geojson", "SpatialPolygonsDataFrame", sp_to_geojson)
 setMethod("as.geojson", "SpatialPolygons", function(x){
   as.geojson(methods::as(x, "SpatialPolygonsDataFrame"))
 })
+
+
+#' @rdname as.geojson
+setMethod("as.geojson", "sf", function(x){
+  as.geojson(sf::as_Spatial(x))
```
Comment (sckott) [inline R/sp_to_geojson.R]: at the top of this method, add a check for sf pkg

```r
if (!requireNamespace('sf')) stop('please install `sf`')
```

**Review by sckott — CHANGES_REQUESTED** [review]
Can you also:

- add `sf` to Suggests in DESCRIPTION file
- add an example to `as.geojson` inside a conditional block

**Comment by sckott** [comment]
thanks for the changes @cpsievert 

not sure if we can have the `setMethod` call for a package in Suggests, looking into that

**Review by sckott — APPROVED** [review]

**Comment by sckott** [comment]
@cpsievert so with help from Jim Hester, I think we should change to removing https://github.com/ropensci/geojson/blob/5fd48ae1a67aec66eec5c97f3576e9e6d361b2a0/R/sp_to_geojson.R#L90-L94 completely from `R/sp_to_geojson.R`

and changing `R/onload.R` to have

```r
.onLoad <- function(libname, pkgname) {
  linting_opts()

  if (requireNamespace('sf', quietly = TRUE)) {
    setMethod("as.geojson", "sf", function(x) {
      as.geojson(sf::as_Spatial(x))
    })
  }
}
```

the only problem then is that the method is not in the docs for `as.geojson` - but users will know its there from the example you added and via `showMethods("as.geojson")`

**Comment by sckott** [comment]
Thanks @cpsievert - looks good. Do you need this on CRAN to be able to use it, or is github version fine?

**Comment by cpsievert** [comment]
No, github is fine, thanks!

**Comment by sckott** [comment]
@cpsievert this is just accepted on CRAN now - binaries soon

**Pull request merged**